web/src/components/User/Login/Login.tsx

```diff
@@ -1,192 +1,252 @@
+            set_user(storedUser)
+            navigate_to_main_page()
         }
-    }
 
-    function storeEmailInput(event: any) {
-        setEmailOriginalInput(event.target.value)
-        setEmailInput(event.target.value)
-    }
+        if (user) {
+            navigate_to_main_page()
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    /** handles successful login
+     */
+    const onAuthSuccessHandler = (name: string, email: string) => {
+        const user: IUser = { name: name, email: email }
+        set_user(user)
+        sessionStorage.setItem('user', JSON.stringify(user))
 
-    function storeCodeInput(event: any) {
-        setCodeInput(event.target.value)
+        navigate_to_main_page()
     }
 
-    function sendEmailCode() {
-        let emailReg = new RegExp('^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.(edu)$')
-        if (emailOriginalInput == '?') {
-            errorAlert('请填写邮箱地址')
-        } else if (!emailReg.test(emailOriginalInput)) {
-            errorAlert('请使用@berkeley.edu邮箱登录')
-        } else {
-            setShowSpan(!showSpan)
-            setButtonHidden(!isButtonHidden)
-            LoginAPI.sendVerificationCode(
-                emailInput,
-                () => console.log('Successfully sent'),
-                (error: any) => {console.log(error); errorAlert('服务器错误，请重试')}
-            )
-            countDown(countDownCurr)
+    /** post request to retrieve auth code after conditions are met
+     */
+    const request_auth_code = () => {
+        const error_handler = () => {
+            set_email_error(true)
+            setTimeout(() => {
+                set_email_error(false)
+            }, 300)
         }
-    }
 
-    function countDown(time: number) {
-        let intervalId = setInterval(() => {
-            setCountDownCurr(prevCount => prevCount - 1)
-            time = time - 1
-            if (time === 0) {
-                clearInterval(intervalId)
-                stopCount()
-            }
-        }, 1000)
-    }
+        // check if cooling down
+        if (auth_cooldown < 60) {
+            console.log(`cooling down, try again in ${auth_cooldown}`)
+            error_handler()
+            return
+        }
 
-    const stopCount = () => {
-        setShowSpan(false)
-        setButtonHidden(false)
-        setCountDownCurr(countDownInit)
-    }
+        // check if email is incomplete
+        if (!email_address.toLowerCase().match(/^[^@]+@[a-z.]+\.edu/)) {
+            console.log('wrong email format')
+            error_handler()
+            return
+        }
 
-    const onAuthenticateSuccess = (response: any) => {
-        console.log(response)
-        saveDataToLocalStorage(emailInput, response['access_token'])
-        navigate('/dashboard')
+        // initiate 60s cooldown
+        start_cooldown()
+
+        // post request
+        LoginAPI.sendVerificationCode(
+            email_address,
+            () => {},
```

Comment:
Should we display some kind of message here saying that the code is sent out successfully?

Reply:
fixed in commit "Added confirmation animation to auth code button"